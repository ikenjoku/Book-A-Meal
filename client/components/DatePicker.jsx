import React, { Component } from 'react';
import moment from 'moment';


export default class DatePicker extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedDate: '',
      formattedDate: '',
    }
  }
  onDateChange = (event) => {
    const selectedDate = event.target.value;
    const formattedDate = moment(selectedDate).format("dddd, MMMM Do YYYY");

    this.setState(() => (
      {
        selectedDate: selectedDate,
        formattedDate: formattedDate,
      }
    ));
    if (this.props.onChange) {
      this.props.onChange({ selectedDate })
    }
  }
  onFormSubmit = (event) => {
    event.preventDefault();
    const { selectedDate, formattedDate } = this.state;
    this.props.onSubmit({ selectedDate, formattedDate })
  }

  render() {
    return (
      <div className="order-history-date-picker center">
        <form onSubmit={this.onFormSubmit}>
          <input
            className="input-field-control"
            type="date"
            name="menu-date"
            value={this.state.selectedDate}
            onChange={this.onDateChange}
          />
          <input
            className="input-btn-control"
            type="submit"
            value={this.props.btnName} />
        </form>
      </div>
    );
  }
}