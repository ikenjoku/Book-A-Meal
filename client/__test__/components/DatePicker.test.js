import React from 'react';
import { shallow } from 'enzyme';
import DatePicker from '../../components/DatePicker';

describe('DatePicker Component', () => {
  let wrapper, onChange, onSubmit;
  onChange = jest.fn();
  onSubmit = jest.fn();

  test('should render DatePicker correctly', () => {
    wrapper = shallow(
      <DatePicker
        onChange={onChange}
        onSubmit={onSubmit}
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onDateChange when date field is changed', () => {
    wrapper = shallow(
      <DatePicker
        onChange={onChange}
        onSubmit={onSubmit}
      />);

    const onDateChangeSpy = jest.spyOn(
      wrapper.instance(), 'onDateChange'
    );
    const event = {target: {
      value: '2018-06-15',
      name: 'menu-date'
    }}
    wrapper.instance().onDateChange(event);
    expect(onDateChangeSpy).toHaveBeenCalledTimes(1);
  });

  it('should call onFormSubmit when date the form is submitted', () => {
    wrapper = shallow(
      <DatePicker
        onChange={onChange}
        onSubmit={onSubmit}
      />);

    const onFormSubmitSpy = jest.spyOn(
      wrapper.instance(), 'onFormSubmit'
    );
    const event = {preventDefault: jest.fn(0)};
    wrapper.instance().onFormSubmit(event);
    expect(onFormSubmitSpy).toHaveBeenCalledTimes(1);
  });
});
