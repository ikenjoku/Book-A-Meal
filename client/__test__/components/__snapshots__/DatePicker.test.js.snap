// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Component should render DatePicker correctly 1`] = `
<div
  className="order-history-date-picker center"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input-field-control"
      name="menu-date"
      onChange={[Function]}
      type="date"
      value=""
    />
    <input
      className="input-btn-control"
      type="submit"
    />
  </form>
</div>
`;
